answer_unit.php

```diff
@@ -66,20 +73,61 @@ public function __construct() {
         $this->default_mapping = null;
         $this->mapping = null;
         $this->additional_rules = '';
+        $this->part_unit = '';
     }
 
+    /**
+     * Parse a unit into prefix and SI unit
+     * 
+     * @param string $unit get the unit with prefix and SI unit, must be trimmed before
+     * @return string SI unit only
+     */
+    public function parse_prefix_unit($unit) {
+        if (is_array($unit)) {
+            throw new Exception('parse_prefix_unit does not handle arrays!');
+            return '';
+        }
+        $pattern = '/^([\wµμ]*).*/';  /* get only first word, accept also µ μ */
+        $replacement = '${1}';
+        $shu = preg_replace($pattern, $replacement, $unit);
+        if ( (strlen($shu) < 2) ||
+            (in_array($shu, static::$units_special) )) {
+            return $unit;
+        }
+        foreach (static::$prefix_scale_factors as $i => $prefix) {
+            if (str_starts_with($unit,$i)) {
+                $unit = substr($unit, strlen($i));
+                break;
+            }
+        }
+        return $unit;
+    }
 
     /**
      * It assign default rules to this class. It will also reset the mapping. No exception raised.
      *
      * @param string $default_id id of the default rule. Use to avoid reinitialization same rule set
      * @param string $default_rules default rules
      */
-    public function assign_default_rules($default_id, $default_rules) {
-        if ($this->default_id == $default_id) {
+    public function assign_default_rules($default_id, $default_rules, $part_unit = '') {
+        if (($default_id == 2) && ($part_unit != $this->part_unit)) {
```

Comment:
I think it would probably be better to have a class constant `ALL_SI_UNITS = 2` and use it here. This would make the code easier to understand in a few years.